src/app/pages/swap/components/swap-container.tsx

```diff
@@ -0,0 +1,24 @@
+import { Outlet } from 'react-router-dom';
+
+import { Flex } from '@stacks/ui';
+
+import { whenPageMode } from '@app/common/utils';
+import { CENTERED_FULL_PAGE_MAX_WIDTH } from '@app/components/global-styles/full-page-styles';
+
+export function SwapContainer() {
+  return whenPageMode({
+    full: (
+      <Flex
+        maxHeight="90vh"
+        border={['unset', '1px solid']}
+        borderRadius={['unset', '16px']}
+        borderColor={['unset', '#DCDDE2']}
+        maxWidth={['100%', CENTERED_FULL_PAGE_MAX_WIDTH]}
+        minWidth={['100%', CENTERED_FULL_PAGE_MAX_WIDTH]}
+      >
+        <Outlet />
+      </Flex>
+    ),
+    popup: <Outlet />,
+  });
```

Comment:
IMO this should be `SwapContainerLayout`, and `SwapContainer` probably be the biggest component in the PR.

This work is similar to Ledger work where I'm taking a complex flow of many routes and trying to reuse it for different use cases. It's difficult to do this when logic (much of which is use-case specific), is spread throughout a load of different screens.

I suggest looking into creating a context here, and defining the core actions of the feature within this container. It could contain methods that are likely to change between use-case. Let's consider `broadcastTransaction`, this is going to be different between magic and sBTC say.

If this method is in the container, along with the other use-case specific methods, then it'll be really easy for us to create two state trees with custom `SwapContainers` that feed logic down to the child routes of this tree. It's much easier to do this with one component, than having all the logic between the individual pages.

I'd really love to move the send form structure to this as well. I don't think logic to broadcast a tx belongs on the review screen, just because that's where the button is.

Hope this makes sense, and happy to jump on a call and run through how I'm doing it with Ledger as a comparison.

Reply:
Yep, makes sense now looking at it all. I prob should have put this up as a draft, but felt teams were pushing for something to work with. Happy to refactor it.